src/click/decorators.py

```diff
@@ -370,7 +371,8 @@ def callback(ctx, param, value):
             )
 
         echo(
-            message % {"prog": prog_name, "package": package_name, "version": version},
+            _(message)
```

Comment:
This isn't a string literal, so it can't be wrapped. The default is up in the function signature, which means it will be evaluated at import time, which is probably before the application sets up translations. So to make this lazy, we need to set `message=None` in the signature, then do `if message is None: message = _("...")`.

Check if there are any other default arguments that might need this same treatment.